crates/turbopack-ecmascript-plugins/src/transform/swc_ecma_transform_plugins.rs

```diff
@@ -0,0 +1,223 @@
+                    },
+                    util::take::Take,
+                },
+                ecma::ast::Module,
+                plugin::proxies::{HostCommentsStorage, COMMENTS},
+                plugin_runner::plugin_module_bytes::PluginModuleBytes,
+            };
+
+            let mut plugins = vec![];
+            for (plugin_module, config) in &self.plugins {
+                let plugin_module = &plugin_module.await?.0;
+
+                plugins.push((
+                    plugin_module.get_module_name().to_string(),
+                    config.clone(),
+                    Box::new(plugin_module.clone()),
+                ));
+            }
+
+            let should_enable_comments_proxy =
+                !ctx.comments.leading.is_empty() && !ctx.comments.trailing.is_empty();
+
+            //[TODO]: as same as swc/core does, we should set should_enable_comments_proxy
+            // depends on the src's comments availability. For now, check naively if leading
+            // / trailing comments are empty.
+            let comments = if should_enable_comments_proxy {
+                // Plugin only able to accept singlethreaded comments, interop from
+                // multithreaded comments.
+                let mut leading =
+                    swc_core::common::comments::SingleThreadedCommentsMapInner::default();
+                ctx.comments.leading.as_ref().into_iter().for_each(|c| {
+                    leading.insert(c.key().clone(), c.value().clone());
+                });
+
+                let mut trailing =
+                    swc_core::common::comments::SingleThreadedCommentsMapInner::default();
+                ctx.comments.trailing.as_ref().into_iter().for_each(|c| {
+                    trailing.insert(c.key().clone(), c.value().clone());
+                });
+
+                Some(SingleThreadedComments::from_leading_and_trailing(
+                    Rc::new(RefCell::new(leading)),
+                    Rc::new(RefCell::new(trailing)),
+                ))
+            } else {
+                None
+            };
+
+            let transformed_program =
+                COMMENTS.set(&HostCommentsStorage { inner: comments }, || {
+                    let module_program =
+                        std::mem::replace(program, Program::Module(Module::dummy()));
+                    let module_program =
+                        swc_core::common::plugin::serialized::VersionedSerializable::new(
+                            module_program,
+                        );
+                    let mut serialized_program =
+                        PluginSerializedBytes::try_serialize(&module_program)?;
+
+                    let transform_metadata_context = Arc::new(TransformPluginMetadataContext::new(
+                        Some(ctx.file_name_str.to_string()),
+                        //[TODO]: Support env-related variable injection, i.e process.env.NODE_ENV
```

Comment:
Do we have an issue tracking this?

Reply:
no, this is actually more close to swc upstream issue.